src/hal/utils/halcmd_commands.cc

```diff
@@ -2401,6 +2441,14 @@ static const char *data_value(int type, void *valptr)
 	snprintf(buf, 14, "  0x%08lX", (unsigned long)*((hal_u32_t *) valptr));
 	value_str = buf;
 	break;
+    case HAL_S64:
+    snprintf(buf, 14, "  %10" PRId64, (int64_t)*((hal_s64_t *) valptr));
+    value_str = buf;
+    break;
+    case HAL_U64:
+    snprintf(buf, 14, "  0x%08" PRIX64, (uint64_t)*((hal_u64_t *) valptr));
```

Comment:
>This should be 0x%016, since u64 has 8 bytes (16 nybbles). 

I didn't want to reserve that much space in case we have that big number. The number will not be cut, so I tend to let it so.

> Which also means a 14 byte string isn't big enough to hold the result.

Thats true, good catch :+1:

Reply:
> This should be `0x%016`, since u64 has 8 bytes (16 nybbles). 

If we would do this, then also the header of the hal show command needs to be changed as well as the width of all other data types.
This would probably break some tests if there are any which just compare the halcmd output.

If increasing the width of s64 only, it would look like this:
```
$ halcmd show pin abs-s64.0.
Component Pins:
Owner   Type  Dir         Value  Name
    40  s64   IN                        0  abs-s64.0.in
    40  bit   OUT         FALSE  abs-s64.0.is-negative
    40  bit   OUT         FALSE  abs-s64.0.is-positive
    40  s64   OUT                       0  abs-s64.0.out
    40  bit   OUT         FALSE  abs-s64.0.sign
    40  s32   OUT           414  abs-s64.0.time
```

Reply:
The other solution would be to have an ugly formatting only for the rare cases when we have a s64 and it really needs more space as an s32:

```
Component Pins:
Owner   Type  Dir         Value  Name
    40  s64   IN              0  abs-s64.0.in
    40  bit   OUT         FALSE  abs-s64.0.is-negative
    40  bit   OUT         FALSE  abs-s64.0.is-positive
    40  s64   OUT             0  abs-s64.0.out
    40  bit   OUT         FALSE  abs-s64.0.sign
    40  s32   OUT           514  abs-s64.0.time

$ halcmd setp abs-s64.0.in -9223372036854775807
$ halcmd show pin abs-s64.0.
Component Pins:
Owner   Type  Dir         Value  Name
    40  s64   IN     -9223372036854775807  abs-s64.0.in
    40  bit   OUT          TRUE  abs-s64.0.is-negative
    40  bit   OUT         FALSE  abs-s64.0.is-positive
    40  s64   OUT    9223372036854775807  abs-s64.0.out
    40  bit   OUT          TRUE  abs-s64.0.sign
    40  s32   OUT           364  abs-s64.0.time
```

Reply:
I prefer the constant width columns, sized to fit the largest possible string.  Hopefully any test that parses halcmd output does it in a non-crazy way, so it would deal gracefully with this change.

But i'm not doing this work, you are, and i trust your judgement on the layout issue.